lib/Timber.php

```diff
@@ -124,67 +124,164 @@ class_alias( 'Timber\Timber', 'Timber' );
 	 *
-	 * @return array|bool|null
+	 * @api
+	 * @example
+	 * ```php
+	 * // Using the global query.
+	 * $posts = Timber::get_posts();
+	 *
+	 * // Using the WP_Query argument format.
+	 * $posts = Timber::get_posts( [
+	 *     'post_type'     => 'article',
+	 *     'category_name' => 'sports',
+	 * ] );
+	 * ```
+	 * @todo Remove warnings in Timber 3.0
+	 *
+	 * @param array|false $query   Optional. A WordPress-style query. Use an array in the same way
+	 *                             you would use the `$args` parameter in
+	 *                             [WP_Query](https://developer.wordpress.org/reference/classes/wp_query/).
+	 *                             Passing no parameter of `false` will use the current global
+	 *                             query. Default `false`.
+	 * @param array       $options {
+	 *     An array of options for the query.
+	 *
+	 *     @type bool $merge_default Optional. Whether to merge the arguments passed in `query` with
+	 *                               the default arguments that WordPress uses for the current
+	 *                               template. Default `false`.
+	 * }
+	 *
+	 * @return PostCollection|bool
 	 */
-	public static function get_posts( $query = false, $PostClass = 'Timber\Post', $return_collection = false ) {
-		return PostGetter::get_posts($query, $PostClass, $return_collection);
+	public static function get_posts( $query = false, $options = [] ) {
+		if ( is_string( $query ) ) {
+			// @todo Add link to documentation for get_posts().
+			Helper::doing_it_wrong(
+				'Timber::get_posts()',
+				"Querying posts by using a query string was removed in Timber 2.0. Pass in the query string as an options array instead. For example, change Timber::get_posts( 'post_type=portfolio&posts_per_page=3') to Timber::get_posts( [ 'post_type' => 'portfolio', 'posts_per_page' => 3 ] ).",
+				'2.0.0'
+			);
+		}
+
+		if ( is_string( $options ) ) {
+			// @todo Add link to Class Maps documentation.
+			Helper::doing_it_wrong(
+				'Timber::get_posts()',
+				'The $PostClass parameter for passing in the post class to use in Timber::get_posts() was removed in Timber 2.0. Use Class Maps (LINK HERE) instead.',
+				'2.0.0'
+			);
+		}
+
+		if ( 3 === func_num_args() ) {
+			Helper::doing_it_wrong(
+				'Timber::get_posts()',
+				'The $return_collection parameter to control whether a post collection is returned in Timber::get_posts() was removed in Timber 2.0.',
+				'2.0.0'
+			);
+		}
+
+		/**
+		 * @todo Define all default $options.
+		 * @todo Actually apply options.
+		 */
+		$options = wp_parse_args( $options, [
+			'merge_default' => false,
+		] );
+
+		// @todo Use factory.
+		return PostGetter::get_posts( $query, 'Timber\Post', true );
 	}
 
 	/**
 	 * Query post.
 	 *
 	 * @api
-	 * @deprecated since 2.0.0 Use `new Timber\Post()` instead.
+	 * @deprecated since 2.0.0 Use `Timber::get_post()` instead.
 	 *
-	 * @param mixed  $query
-	 * @param string $PostClass
+	 * @param null|int|\WP_Post $post
```

Comment:
this should change from `null` to `false`

Reply:
Resolved in a2abbb38096cd5abc762c4827c37d037b1215a78.